ext/standard/string.c

```diff
@@ -1026,7 +1026,11 @@ PHPAPI void php_implode(const zend_string *glue, HashTable *pieces, zval *return
 		}
 
 		cptr -= ZSTR_LEN(glue);
-		memcpy(cptr, ZSTR_VAL(glue), ZSTR_LEN(glue));
+		if (ZSTR_LEN(glue) == 1) {
```

Comment:
This is in loop, is `ZSTR_LEN(glue)` guaranteed to be evaluated only once?

What about moving the condition outside the loop, ie. "duplicating" the loop body?

Also what about the case when the glue is empty string? (we use that strongly in https://github.com/atk4/ui/blob/6.0.0/src/HtmlTemplate.php#L570 - with 10k calls per webpage request easily)

Reply:
> This is in loop, is ZSTR_LEN(glue) guaranteed to be evaluated only once?

No. But chances are that it doesn't matter if this is in a variable or not because you'll need a load from a spill slot in the variable case most likely.